src/Faker/Provider/sv_SE/Person.php

```diff
@@ -132,17 +132,40 @@ public function personalIdentityNumber(\DateTime $birthdate = null, $gender = nu
             $birthdate = \Faker\Provider\DateTime::dateTimeThisCentury();
         }
         $datePart = $birthdate->format('ymd');
+        $randomDigits = $this->getBirthNumber($gender);
 
+        $checksum = Luhn::computeCheckDigit($datePart . $randomDigits);
+
+        return $datePart . '-' . $randomDigits . $checksum;
+    }
+
+    /**
+     * @param string $gender Person::GENDER_MALE || Person::GENDER_FEMALE
+     *
+     * @return string of three digits
+     */
+    protected function getBirthNumber($gender = null)
+    {
         if ($gender && $gender == static::GENDER_MALE) {
```

Comment:
Can we make the checks type safe? I agree that we can force a string due to a breaking change but we could force the type with an ===

Please note that there are multiple occurrences in this class

Reply:
Should I fix the whole class or just this method?

Reply:
It was only in this method. It's fixed now.